code/datums/components/bubble_icon_override.dm

```diff
@@ -0,0 +1,38 @@
+/datum/component/bubble_icon_override
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	can_transfer = TRUE //sure why not
+	///The override to the default bubble icon for the atom
+	var/bubble_icon
+	///The priority of this bubble icon compared to others
+	var/priority
+
+/datum/component/bubble_icon_override/Initialize(bubble_icon, priority)
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	src.bubble_icon = bubble_icon
+	src.priority = priority
+
+/datum/component/bubble_icon_override/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_GET_BUBBLE_ICON, PROC_REF(return_bubble_icon))
+	get_bubble_icon()
+
+/datum/component/bubble_icon_override/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_GET_BUBBLE_ICON)
+	get_bubble_icon()
```

Comment:
this either should not be a component or should be reacting to more diverse signals, this is not an event these are proc calls.